lib/liquid.lua

```diff
@@ -2547,28 +2549,37 @@ end
 -------------------------------------------------------------ParserContext end ---------------------------------------------------------
 -------------------------------------------------------------FileSystem begin ---------------------------------------------------------
 -- local FileSystem = {}
-function FileSystem:new( location )
+function FileSystem:new( get, error_handler )
     -- body
     local instance = {}
     setmetatable(instance, {__index = FileSystem})
-    instance.location = location
+    instance.get = get
     instance.text = nil
+    instance.error_handler = error_handler
     return instance
 end
 --
-function FileSystem:genertic_get( ... )
+function FileSystem:generic_get( location )
+    local error_handler = self.error_handler
     -- body
     if self.get and type(self.get) == "function" then
-        local status, err = pcall(self.get, self.location)
-        if status then
-            return err
+        local ok, ret = pcall(self.get, location)
+
+        if ok and ret then
+            return tostring(ret)
+        elseif ok then
+            return error_handler(location, "cannot render empty template" )
         else
-            error("get template: " .. location .. " fail by user self defined get method" .. tostring(err))
+            return error_handler(location, ret)
```

Comment:
I think it is better to just pass it along. User can throw whatever as an error (for example table). Then serialising it to string could do harm.